packages/jaeger-ui/src/components/SearchTracePage/index.jsx

```diff
@@ -309,4 +310,4 @@ function mapDispatchToProps(dispatch) {
   };
 }
 
-export default connect(mapStateToProps, mapDispatchToProps)(SearchTracePageImpl);
+export default withRouteProps(connect(mapStateToProps, mapDispatchToProps)(SearchTracePageImpl));
```

Comment:
Curious, why wasn't this change needed in the previous PR where withRouteProps was introduced? Does it work both ways?

Reply:
No, it doesn't work both ways.

In the previous PR's code, it was working fine. I have just verified with the code on the main branch.

This file uses: `this.props.history.push` and even in the previous PR, I didn't see it using `withRouter`.
So, where did its `history` prop come from in the main branch? That's strange.

Reply:
You can too verify this by clicking on a trace spot from this chart:

![image](https://github.com/jaegertracing/jaeger-ui/assets/94157520/5a7dc565-05df-4719-9e9d-b3b87107963a)

In the main branch's code, it was working.

But in this PR's code, without the use of `withRouteProps`, this doesn't seem to work. I couldn't figure out why.

Reply:
Just figured it out.

Actually, it is due to the change in how we are using `Route children` and `Route component`.